release/src/main/scripts/script.config

```diff
@@ -0,0 +1,139 @@
+#    this work for additional information regarding copyright ownership.
+#    The ASF licenses this file to You under the Apache License, Version 2.0
+#    (the "License"); you may not use this file except in compliance with
+#    the License.  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# This file contains all configurations required by release scripts.
+
+
+########################################
+# Register all configs defined in this file.
+########################################
+
+CONFIG_NAME_LIST=(
+  RELEASE_VER
+  RC_NUM
+  REPO_URL
+  INSTALL_GCLOUD
+  USER_GCP_PROJECT
+  USER_GCS_BUCKET
+  USER_SERVICE_ACCOUNT_EMAIL
+  INSTALL_HUB
+  USER_REMOTE_URL
+  GITHUB_USERNAME
+  INSTALL_GNOME_TERMINAL
+  LOCAL_BEAM_DIR
+  java_quickstart_direct
+  java_quickstart_apex_local
+  java_quickstart_flink_local
+  java_quickstart_spark_local
+  java_quickstart_dataflow
+  java_mobile_game
+  python_quickstart_mobile_game
+  python_leaderboard_direct
+  python_leaderboard_dataflow
+  python_gamestats_direct
+  python_gamestats_dataflow
+)
+
+
+########################################
+# Environment Configurations
+########################################
+
+# Beam version of current release
+# e.g. 2.14.0
+RELEASE_VER=
+
+# Release candidate number
+# This is an identifier for each candidate we built for release. Start from 1
+# and increment if new candidate is built.
+# e.g. 1
+RC_NUM=
+
+# The repo URL from the vote email sent by Release Manager
+# e.g. https://repository.apache.org/content/repositories/orgapachebeam-0000
+REPO_URL=
+
+# Install Google Cloud SDK
+# Google Cloud SDK is required to run validation pipeline on DataflowRunner.
+# Set to true so that it will be installed if not found from local.
+INSTALL_GCLOUD=true
+
+# GCP project id
+# Required for running pipeline with DataflowRunner for validation.
+# e.g. apache-beam-testing
+USER_GCP_PROJECT=
+
+# GCS bucket name
+# Required for running pipeline with DataflowRunner for validation.
+# e.g. bucket-name (do not include 'gs://')
+USER_GCS_BUCKET=
+
+# GCP service account email
+# This service account should be under your project ${USER_GCP_PROJECT}. If you
+# don't have one, create it from GCP IAM console and set as project owner. You
+# can leave it empty if you have gcloud setup before.
+USER_SERVICE_ACCOUNT_EMAIL=
+
+# GCP application credentials
+# This is used to setup gcloud. Override following value if you already set
+# it in bashrc.
+GOOGLE_APPLICATION_CREDENTIALS=~/.config/gcloud/application_default_credentials.json
```

Comment:
I think this should be checked by a script - if a user has this variable set and whether this file exists, and crash if it doesn't

Reply:
I take closer look at google credential and realized this parameter can be removed if USER_SERVICE_ACCOUNT_EMAIL is provided. It's not likely people have the credential file in their machine while gcloud is not installed. We can always create it by providing USER_SERVICE_ACCOUNT_EMAIL. 

So I would do following as you suggested but without this parameter:
1. use GOOGLE_APPLICATION_CREDENTIALS if it's already exported in bashrc.
2. if USER_SERVICE_ACCOUNT_EMAIL is provided, create a credential file and have GOOGLE_APPLICATION_CREDENTIALS points to it.
3. otherwise, crash the script and let people create USER_SERVICE_ACCOUNT_EMAIL first.